airbyte-cdk/python/airbyte_cdk/sources/connector_state_manager.py

```diff
@@ -52,10 +54,10 @@ def get_stream_state(self, stream_name: str, namespace: Optional[str]) -> Mappin
         """
         stream_state = self.per_stream_states.get(HashableStreamDescriptor(name=stream_name, namespace=namespace))
         if stream_state:
-            return stream_state.dict()
+            return stream_state.dict()  # type: ignore # this returns a dict...
```

Comment:
Does `dict` implement the interface of `MutableMapping`? It feels like we shouldn't ignore something here

Reply:
Updated the comment to clarify that the problem is mypy doesn't properly detect the type of stream_state.dict()